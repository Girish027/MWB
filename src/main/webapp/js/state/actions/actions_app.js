
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
