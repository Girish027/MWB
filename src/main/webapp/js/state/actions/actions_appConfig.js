import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
