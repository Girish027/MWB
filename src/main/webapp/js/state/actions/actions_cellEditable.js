import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
