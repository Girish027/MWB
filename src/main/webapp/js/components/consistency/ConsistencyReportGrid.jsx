import React, { Component } from 'react';
import { connect } from 'react-redux';
import store from 'state/configureStore';
import * as actionsConsistencyReport from 'state/actions/actions_consistency_report';
import * as actionsApp from 'state/actions/actions_app';
import AlertIcon from 'grommet/components/icons/base/Alert';
import FormCheckmarkIcon from 'grommet/components/icons/base/FormCheckmark';
import CellEditableManualTag from 'components/controls/grid/CellEditableManualTag';
import {
  setTag, removeTag, showBulkTagCell, hideBulkTagCell,
} from 'state/actions/actions_datasets_transformed_tag';
import { modalDialogChange } from 'state/actions/actions_app';
import TaggerGrid from 'components/controls/grid/TaggerGrid';
import BulkTagCellEditable from 'components/tagging/grid/BulkTagCellEditable';
import SimpleDialog from 'components/controls/SimpleDialog';
import { getLanguage } from 'state/constants/getLanguage';
import Constants from 'constants/Constants';

const languageConstants = getLanguage();
const { DISPLAY_MESSAGES } = languageConstants;
import {
  Xmark,
  Tag,
  ContextualActionsBar,
  ContextualActionItem,
} from '@tfs/ui-components';

export const stateKey = 'ConsistencyReportGridState-v1';

export class ConsistencyReportGrid extends Component {
  constructor(props) {
    super(props);
    this.props = props;

    this.handleApplySuggestedClick = this.handleApplySuggestedClick.bind(this);
    this.handleBulkTagClick = this.handleBulkTagClick.bind(this);
    this.handleBulkUnTagClick = this.handleBulkUnTagClick.bind(this);
    this.handleOnClickOk = this.handleOnClickOk.bind(this);

    this.state = {
      selectedIndexes: [],
      selectedTranscriptionHashes: [],
      selectedConflictRowsWithSuggestedIntent: [],
      selectedSuggestedIntents: [],
      bulkTagFormVisible: false,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.props = nextProps;
    const { onTagsChange, dispatch } = this.props;
    const {
      projectId, searchResults, showBulkTag, updateBulkTag,
    } = this.props.consistencyReport;

    if (!searchResults.length) {
      this.setState({
        selectedIndexes: [],
        selectedTranscriptionHashes: [],
        selectedConflictRowsWithSuggestedIntent: [],
        selectedSuggestedIntents: [],
        bulkTagFormVisible: false,
      });
      return;
    }

    const selectedIndexes = this.getSelectedIndexes();
    const selectedTranscriptionHashes = [];
    const selectedConflictRowsWithSuggestedIntent = [];
    const selectedSuggestedIntentsObj = {};

    selectedIndexes.forEach((index) => {
      const row = typeof searchResults[index] !== 'undefined' ? searchResults[index] : {};
      if (!row.transcriptionHash) {
        return;
      }
      selectedTranscriptionHashes.push(row.transcriptionHash);
      if (row.intentConflict && row.suggestedIntent) {
        selectedConflictRowsWithSuggestedIntent.push(row);
        selectedSuggestedIntentsObj[row.suggestedIntent] = true;
      }
    });

    this.setState({
      selectedIndexes,
      selectedTranscriptionHashes,
      selectedConflictRowsWithSuggestedIntent,
      selectedSuggestedIntents: Object.keys(selectedSuggestedIntentsObj),
      bulkTagFormVisible: showBulkTag && selectedTranscriptionHashes.length,
    });

    if (showBulkTag && !selectedTranscriptionHashes.length) {
      dispatch(hideBulkTagCell());
    }
    if (showBulkTag && selectedTranscriptionHashes.length && updateBulkTag) {
      dispatch(setTag({
        projectId,
        hashList: selectedTranscriptionHashes,
        intent: updateBulkTag,
        isUpdate: true,
      })).then(() => {
        dispatch(hideBulkTagCell());
        if (onTagsChange) {
          onTagsChange();
        }
      });
    }
  }

  handleOnClickOk(transcriptionHashesBySuggestedIntent) {
    const { dispatch, onTagsChange, consistencyReport } = this.props;
    const { projectId } = consistencyReport;

    let promises = [],
      suggestedIntent;
    for (suggestedIntent in transcriptionHashesBySuggestedIntent) {
      if (!transcriptionHashesBySuggestedIntent.hasOwnProperty(suggestedIntent)) {
        continue;
      }
      promises.push(dispatch(setTag({
        projectId,
        hashList: transcriptionHashesBySuggestedIntent[suggestedIntent],
        intent: suggestedIntent,
        isUpdate: true,
      })));
    }

    Promise.all(promises)
      .then(() => {
        if (onTagsChange) {
          onTagsChange();
        }
      })
      .catch(() => {
      });

    dispatch(modalDialogChange(null));
  }

  getSelectedIndexes() {
    const { selection } = this.props;
    const currentSelection = selection.get(this.config.stateKey);
    if (currentSelection) {
      const selectionData = currentSelection.toJS();
      if (selectionData.indexes) {
        return [...selectionData.indexes];
      }
    }
    return [];
  }

  handleApplySuggestedClick() {
    const { dispatch } = this.props;
    const { selectedConflictRowsWithSuggestedIntent } = this.state;

    let transcriptionHashesBySuggestedIntent = {},
      differentSuggestedCount = 0,
      lastSuggested = '';

    if (selectedConflictRowsWithSuggestedIntent.length) {
      selectedConflictRowsWithSuggestedIntent.forEach((row) => {
        if (!row.suggestedIntent || !row.transcriptionHash) {
          return;
        }
        if (!transcriptionHashesBySuggestedIntent[row.suggestedIntent]) {
          transcriptionHashesBySuggestedIntent[row.suggestedIntent] = [];
          differentSuggestedCount++;
          lastSuggested = row.suggestedIntent;
        }
        transcriptionHashesBySuggestedIntent[row.suggestedIntent].push(row.transcriptionHash);
      });

      dispatch(modalDialogChange({
        type: Constants.DIALOGS.SIMPLE_DIALOG,
        header: 'Apply Suggested Intents',
        message: (
          <div>
            <div>
              You are about to tag
              {' '}
              <b>{selectedConflictRowsWithSuggestedIntent.length}</b>
              {' '}
              of transcriptions with
              {differentSuggestedCount < 2 ? (
                <span>
                  {' '}
                  tag "
                  {lastSuggested}
                  ".
                </span>
              ) : (
                <span>
                  <b>
                    {' '}
                    {differentSuggestedCount}
                  </b>
                  {' '}
                  corresponding suggested tags.
                </span>
              )}
            </div>
            <p>
              Are you sure?
            </p>
          </div>
        ),
        actions: [SimpleDialog.ACTION_OK, SimpleDialog.ACTION_CANCEL],
        className: 'ConsistencyReportConfirmApplySuggested',
        onOk: () => this.handleOnClickOk(transcriptionHashesBySuggestedIntent),
      }));
    }
  }

  handleBulkTagClick() {
    const { selectedIndexes } = this.state;

    if (selectedIndexes.length) {
      this.props.dispatch(showBulkTagCell());
    }
  }

  handleBulkUnTagClick() {
    const { consistencyReport, onTagsChange, dispatch } = this.props;
    const { selectedTranscriptionHashes, selectedIndexes } = this.state;
    const { projectId } = consistencyReport;

    if (selectedIndexes.length) {
      dispatch(removeTag({ projectId, hashlist: selectedTranscriptionHashes }))
        .then(() => {
          if (onTagsChange) {
            onTagsChange();
          }
        })
        .catch(() => {
        });
    }
  }

  render() {
    const { paddingTop } = this.props;
    const {
      selectedIndexes, selectedConflictRowsWithSuggestedIntent,
      selectedSuggestedIntents, bulkTagFormVisible,
    } = this.state;
    const { isSearching, isError, projectId } = this.props.consistencyReport;
    const style = {};
    if (isSearching || isError) {
      style.display = 'none';
    }
    if (paddingTop) {
      style.paddingTop = `${paddingTop}px`;
    }

    const bulkCellData = {
      column: null,
      value: 0,
      row: { manualTag: '' },
      key: 0,
      index: 0,
    };

    return (
      <div
        id="ConsistencyReportGridContainer"
        style={style}
      >
        <div
          id="ConsistencyReportGridActions"
          direction="row"
          style={{
          }}
        >
          {/* <Box flex>
            <div className="PageInfo">
                            Page {currentPage} <span className="GrayText">of {maxPage} [{total} Search Result{total == 1 ? '' : 's'}]</span>
                            &nbsp;
              {selectedIndexes.length ? <span>
                {selectedIndexes.length} Row{selectedIndexes.length == 1 ? '' : 's'} Selected
              </span> : null}
            </div>
          </Box> */}
          <div className="ActionsContainer" direction="row">
            {bulkTagFormVisible
              ? (
                <div className="bulkTagLabelCellContainer">
                  <span className="bulkTagLabel">Bulk Tag:&nbsp;</span>
                  <BulkTagCellEditable cellData={bulkCellData} projectId={projectId} suggestedIntents={selectedSuggestedIntents} />
                </div>
              )
              : (
                <ContextualActionsBar>
                  <ContextualActionItem
                    onClickAction={this.handleApplySuggestedClick}
                    icon={Tag}
                    disabled={!selectedConflictRowsWithSuggestedIntent.length}
                    styleOverride={{ paddingLeft: '20px' }}
                  >
                    APPLY SUGGESTED
                  </ContextualActionItem>
                  <ContextualActionItem
                    disabled={!selectedIndexes.length}
                    onClickAction={this.handleBulkTagClick}
                    icon={Tag}
                  >
                    BULK TAG
                  </ContextualActionItem>
                  <ContextualActionItem
                    disabled={!selectedIndexes.length}
                    onClickAction={this.handleBulkUnTagClick}
                    icon={Xmark}
                  >
                    BULK UNTAG
                  </ContextualActionItem>
                </ContextualActionsBar>
              )
            }
          </div>
        </div>

        <div>
          <TaggerGrid
            id="ConsistencyReportGrid"
            className="ConsistencyReportGrid"
            config={this.config}
          />
        </div>
      </div>
    );
  }

  get config() {
    const { dispatch } = this.props;
    const height = '';

    const stateful = true;

    const emptyDataMessage = 'No Search Results';

    const columns = [
      {
        name: 'Count',
        width: '5%',
        dataIndex: 'documentCount',
        sortable: true,
        hideable: true,
        resizable: false,
        hidden: true,
      },
      {
        name: 'Has Conflict',
        width: '3%',
        dataIndex: 'intentConflict',
        sortable: true,
        hideable: true,
        resizable: false,
        renderer: ({ column, value, row }) => (value
          ? <span className="HasConflictIconContainer"><AlertIcon /></span>
          : <span className="NoConflictIconContainer"><FormCheckmarkIcon /></span>),
      },
      {
        name: 'Text String',
        width: '20%',
        dataIndex: 'textStringForTagging',
        sortable: true,
        hideable: false,
      },
      {
        name: 'Normalized Form',
        width: '15%',
        dataIndex: 'normalizedForm',
        sortable: true,
        hideable: true,
      },
      {
        name: 'Group',
        width: '10%',
        dataIndex: 'normalizedFormGroup',
        sortable: true,
        hideable: true,
      },
      {
        name: 'Original Intent',
        width: '15%',
        dataIndex: 'intent',
        sortable: true,
        hideable: true,
        renderer: ({ column, value, row }) => {
          const state = store.getState();
          const { consistencyReport } = state;
          const { projectId } = consistencyReport;

          return (
            <CellEditableManualTag
              stateKey="ConsistencyReport"
              value={row.intent}
              rowIndex={row._rowIndex}
              columnIndex={0}
              maxRowIndex={consistencyReport.searchResults ? consistencyReport.searchResults.length - 1 : 0}
              maxColumnIndex={0}
              stopClickPropagation
              projectId={consistencyReport.projectId}
              highlightTags={row.suggestedIntent ? [row.suggestedIntent] : []}
              activateNextCellOnEnter={false}
              onChange={(newValue) => {
                if (newValue.length) {
                  dispatch(setTag({
                    projectId,
                    hashList: [row.transcriptionHash],
                    intent: newValue,
                    isUpdate: true,
                  }))
                    .then(() => {
                      dispatch(actionsApp.displayGoodRequestMessage(DISPLAY_MESSAGES.tagUpdated));
                    });
                } else {
                  dispatch(removeTag({
                    projectId,
                    hashList: [row.transcriptionHash],
                  }))
                    .then(() => {
                      dispatch(actionsApp.displayGoodRequestMessage(DISPLAY_MESSAGES.tagRemoved));
                    });
                }
              }}
            />
          );
        },
      },
      {
        name: 'Suggested Intent',
        width: '15%',
        dataIndex: 'suggestedIntent',
        sortable: true,
        hideable: true,
      },
      {
        name: 'Other Intents With Same Form',
        width: '20%',
        dataIndex: 'intents',
        sortable: true,
        hideable: true,
        renderer: ({ column, value, row }) => (value && value.length ? value.join(', ') : ''),
      },
    ];

    const dataSource = function getData(arg1, arg2, arg3) {
      const state = store.getState();
      const sort = arg3 ? arg3.sort : state.consistencyReport.sort;
      const {
        projectId, query, filter, startIndex, limit,
      } = state.consistencyReport;

      return new Promise((resolve, reject) => {
        dispatch(actionsConsistencyReport.requestSearch({
          projectId,
          query,
          filter,
          sort,
          startIndex,
          limit,
        }))
          .then(() => {
            const state = store.getState();
            resolve({
              data: state.consistencyReport.searchResults,
              total: state.consistencyReport.total,
            });
          })
          .catch((error) => {
            reject(error);
          });
      });
    };

    const plugins = {
      COLUMN_MANAGER: {
        resizable: true,
        moveable: true,
        minColumnWidth: 8,
        sortable: {
          enabled: true,
          method: 'remote',
          sortingSource: dataSource,
        },
      },
      GRID_ACTIONS: {
        iconCls: 'action-icon',
        menu: [],
      },
      SELECTION_MODEL: {
        enabled: true,
        mode: 'checkbox-multi',
        allowDeselect: true,
      },
    };

    const events = {

    };

    return {
      stateKey,
      height,
      stateful,
      columns,
      plugins,
      dataSource,
      emptyDataMessage,
      events,
    };
  }
}

const mapStateToProps = state => ({
  app: state.app,
  consistencyReport: state.consistencyReport,
  selection: state.selection,
});

const mapDispatchToProps = dispatch => ({
  dispatch,
});

export default connect(mapStateToProps, mapDispatchToProps)(ConsistencyReportGrid);
