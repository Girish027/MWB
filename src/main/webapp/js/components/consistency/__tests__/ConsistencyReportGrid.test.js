import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import store from 'state/configureStore';
import ReduxConsistencyReportGrid from 'components/consistency/ConsistencyReportGrid';
import { ConsistencyReportGrid } from 'components/consistency/ConsistencyReportGrid';
import toJSON from 'enzyme-to-json';
import * as datasetsTransformed from 'state/actions/actions_datasets_transformed_tag';
import * as actionsApp from 'state/actions/actions_app';

describe('<ConsistencyReportGrid />', () => {
  let wrapper;

  const consistencyReport = {
    searchResults: [{ transcriptionHash: 'ddsad', intentConflict: true, suggestedIntent: true }],
    updateBulkTag: [],
    projectId: '12',
  };
  const props = {
    paddingTop: 12,
    currentPage: 2,
    maxPage: 12,
    consistencyReport,
    onTagsChange: jest.fn(),
    selection: new Map(),
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<ConsistencyReportGrid
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    beforeAll(() => {
      wrapper = mount(<Provider store={store}>
        <MemoryRouter
          initialEntries={['/']}
          initialIndex={0}
        >
          <ReduxConsistencyReportGrid />
        </MemoryRouter>
      </Provider>);
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly for default props on shallow copy', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      props.dispatch = jest.fn();
      datasetsTransformed.showBulkTagCell = jest.fn(() => 'called showBulkTagCell');
      datasetsTransformed.removeTag = jest.fn(() => 'called removeTag');
      actionsApp.modalDialogChange = jest.fn(() => 'called modalDialogChange');
    });

    describe('handleApplySuggestedClick', () => {
      test('should call handleApplySuggestedClick', () => {
        wrapper = getShallowWrapper(props);
        wrapper.setState({ selectedConflictRowsWithSuggestedIntent: [{ suggestedIntent: 'sdf', transcriptionHash: 'sff' }] });
        wrapper.instance().handleApplySuggestedClick();
      });
    });

    describe('handleBulkTagClick', () => {
      test('should call handleBulkTagClick', () => {
        wrapper = getShallowWrapper(props);
        wrapper.setState({ selectedIndexes: 'aasd' });
        wrapper.instance().handleBulkTagClick();
        expect(datasetsTransformed.showBulkTagCell).toHaveBeenCalledWith();
        expect(props.dispatch).toHaveBeenCalledWith('called showBulkTagCell');
      });
    });

    describe('handleBulkUnTagClick', () => {
      test('should call handleBulkUnTagClick', () => {
        wrapper = getShallowWrapper(props);
        wrapper.setState({ selectedIndexes: '', selectedTranscriptionHashes: ['fdf'] });
        wrapper.instance().handleBulkUnTagClick();
        expect(datasetsTransformed.removeTag).not.toHaveBeenCalled();
      });
    });

    describe('getSelectedIndexes', () => {
      test('should call getSelectedIndexes', () => {
        wrapper = getShallowWrapper(props);
        const result = wrapper.instance().getSelectedIndexes();
        expect(result).toStrictEqual([]);
      });

      test('should call getSelectedIndexes with props', () => {
        props.selection.get = jest.fn().mockReturnValue({ toJS: jest.fn().mockReturnValue({ indexes: [1, 2, 3] }) });
        wrapper = getShallowWrapper(props);
        const result = wrapper.instance().getSelectedIndexes();
        expect(result).toEqual([1, 2, 3]);
      });
    });

    describe('handleOnClickOk', () => {
      test('should call handleOnClickOk', () => {
        const transcriptionHashesBySuggestedIntent = [{ suggestedIntent: 'dsdfdf' }];
        wrapper = getShallowWrapper(props);
        wrapper.instance().handleOnClickOk(transcriptionHashesBySuggestedIntent);
        expect(props.dispatch).toHaveBeenCalledWith('called modalDialogChange');
        expect(actionsApp.modalDialogChange).toHaveBeenCalledWith(null);
      });
    });

    describe('componentWillReceiveProps', () => {
      test('should call componentWillReceiveProps with searchResults', () => {
        const nextProps = {
          ...props,
        };
        wrapper = getShallowWrapper(props);
        wrapper.instance().getSelectedIndexes = jest.fn().mockReturnValue([0]);
        wrapper.instance().componentWillReceiveProps(nextProps);
        expect(wrapper.state().selectedIndexes).toStrictEqual([0]);
      });
    });
  });
});
