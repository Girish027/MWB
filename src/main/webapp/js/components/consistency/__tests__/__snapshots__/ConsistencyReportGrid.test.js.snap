// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsistencyReportGrid /> Snapshots renders correctly for default props on shallow copy 1`] = `
<div
  id="ConsistencyReportGridContainer"
  style={
    Object {
      "paddingTop": "12px",
    }
  }
>
  <div
    direction="row"
    id="ConsistencyReportGridActions"
    style={Object {}}
  >
    <div
      className="ActionsContainer"
      direction="row"
    >
      <withTheme(ContextualActionsBar)>
        <withTheme(ContextualActionItem)
          disabled={true}
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          onClickAction={[Function]}
          styleOverride={
            Object {
              "paddingLeft": "20px",
            }
          }
        >
          APPLY SUGGESTED
        </withTheme(ContextualActionItem)>
        <withTheme(ContextualActionItem)
          disabled={true}
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": [Function],
            }
          }
          onClickAction={[Function]}
        >
          BULK TAG
        </withTheme(ContextualActionItem)>
        <withTheme(ContextualActionItem)
          disabled={true}
          icon={[Function]}
          onClickAction={[Function]}
        >
          BULK UNTAG
        </withTheme(ContextualActionItem)>
      </withTheme(ContextualActionsBar)>
    </div>
  </div>
  <div>
    <TaggerGrid
      className="ConsistencyReportGrid"
      config={
        Object {
          "columns": Array [
            Object {
              "dataIndex": "documentCount",
              "hidden": true,
              "hideable": true,
              "name": "Count",
              "resizable": false,
              "sortable": true,
              "width": "5%",
            },
            Object {
              "dataIndex": "intentConflict",
              "hideable": true,
              "name": "Has Conflict",
              "renderer": [Function],
              "resizable": false,
              "sortable": true,
              "width": "3%",
            },
            Object {
              "dataIndex": "textStringForTagging",
              "hideable": false,
              "name": "Text String",
              "sortable": true,
              "width": "20%",
            },
            Object {
              "dataIndex": "normalizedForm",
              "hideable": true,
              "name": "Normalized Form",
              "sortable": true,
              "width": "15%",
            },
            Object {
              "dataIndex": "normalizedFormGroup",
              "hideable": true,
              "name": "Group",
              "sortable": true,
              "width": "10%",
            },
            Object {
              "dataIndex": "intent",
              "hideable": true,
              "name": "Original Intent",
              "renderer": [Function],
              "sortable": true,
              "width": "15%",
            },
            Object {
              "dataIndex": "suggestedIntent",
              "hideable": true,
              "name": "Suggested Intent",
              "sortable": true,
              "width": "15%",
            },
            Object {
              "dataIndex": "intents",
              "hideable": true,
              "name": "Other Intents With Same Form",
              "renderer": [Function],
              "sortable": true,
              "width": "20%",
            },
          ],
          "dataSource": [Function],
          "emptyDataMessage": "No Search Results",
          "events": Object {},
          "height": "",
          "plugins": Object {
            "COLUMN_MANAGER": Object {
              "minColumnWidth": 8,
              "moveable": true,
              "resizable": true,
              "sortable": Object {
                "enabled": true,
                "method": "remote",
                "sortingSource": [Function],
              },
            },
            "GRID_ACTIONS": Object {
              "iconCls": "action-icon",
              "menu": Array [],
            },
            "SELECTION_MODEL": Object {
              "allowDeselect": true,
              "enabled": true,
              "mode": "checkbox-multi",
            },
          },
          "stateKey": "ConsistencyReportGridState-v1",
          "stateful": true,
        }
      }
      id="ConsistencyReportGrid"
    />
  </div>
</div>
`;
